### ISSUE TITLE
CU - Look into how to create and supply errors in demo mode

### ISSUE THREAD

**Issue by cadibemma** [department-of-veterans-affairs/va-mobile-app#9348]
## Proposed Change

Look into how we could easily create and supply errors in demo mode.

## Why Should We Prioritize?

Currently there isn't an easily written detox solution for implementing errors from a backend. With this change, it will open us up to the availability of error testing in detox and make QA/VQA easier to manage if it's accessible through the developer menu.

## Coding Time Estimation

5
	
## Checklist
- [x] Add the relevant team label (Health, global, design system, API, Qa and Release etc.)
- [ ] Attach to ticket to the relevant Team Tech Debt Epic epic (old frontend engineering epic is no longer in use as each team is managing their code's technical debt and code upkeep work)

**Comment by TKDickson** [comment]
Switching to RB as part of QA rotation

**Comment by rbontrager** [comment]
@Sparowhawk Would it be possible to add another couple options to the backend override option? For example:
- To get the 'we can't load some of your VA appointments' error you have use the errors > source field
- To get errors for just claims or just appeals you have to use the errors > service and the errors > errorDetails field

Also I noticed that overriding any of the errors for /v0/health/locations doesn't do anything in demo mode and that you aren't doing any error overrides for the maintenance window call.  

Let me know if you think these are out of scope and I can make another ticket.

**Comment by Sparowhawk** [comment]
technically yes, it wouldn’t be in the BE errors though as that only checks for the custom fields for the alert. This would go in the other errors section
Just trying to think if it would be better to open up the whole source field to a string you type or check boxes for those 2 particular scenarios
and same for the errorDetails

**Comment by dumathane** [comment]
I think this is a follow up ticket CC: @rbontrager @Sparowhawk. Can we move this one forward and cut a new ticket?

**Comment by Sparowhawk** [comment]
maintenance window currently doesn't have an override because it hasn't been migrated over. That api is currently in the error slice. It also doesn't do anything if there is an error. It just interprets an error as having 0 maintenance windows. This one should be in a later ticket and ideally should be added to this screen when the error slice is converted to RQ

**Comment by rbontrager** [comment]
I can make a ticket for the maintenance windows/enhancement for the thing I mentioned above.  I would like to get an answer on why /v0/health/locations isn't working.  I feel like if it doesn't behave like expected and its not easily fixable we should remove that one from the list and cut a new ticket to add it back in once it fully works.

**Comment by Sparowhawk** [comment]
the locations api does work, when that one errors or returns null the data is marked as none noted for the corresponding fields
we don’t have a specific error screen for that api
so going to leave that one

**Comment by Sparowhawk** [comment]
Appointments would require a lot of rewriting of certain things cause technically it's a success response with an error object attached. Which isn't as simple as throw error so our error component catches it.

This would require a decent amount of work and is out of scope of this ticket. So another ticket is needed for this.

**Comment by rbontrager** [comment]
The last minute change seems to have broken some of the calls.  v0/claims is working but v0/appointments isn't.

**Comment by rbontrager** [comment]
@Sparowhawk Now that I'm using the correct build I'm seeing a bunch of app crashes as I go through testing all the api calls.  I've got an example below but it doesn't always happen on backend override and I've seen it occur when tapping the network/other codes options as well.  This is just the one I know what I did to reproduce. 

1. Go to direct deposit /v0/payment-information/benefits and select backend override. Don't type anything in
2. Go to payments > direct deposit
3. Go back to the payments page
4. Navigate back to direct deposit and try to edit the info for the backend override in direct deposit

**Comment by rbontrager** [comment]
@Sparowhawk 
1. I can't seem to get /v0/messaging/health/messages/recipients to work
2. While going through the api calls for messages I've crashed the app (this time with an error code) multiple times.  Steps to inconsistently reproduce: Select network for v0/messaging/health/messages/. Go to the first message and open it.  Go back to v0/messaging/health/messages/ and uncheck. Notice that it sometimes crashes. This one might be out of scope (but I want to know what is happening) so if it is I'll make another ticket.

**Comment by Sparowhawk** [comment]
Fixed recipients

**Comment by Sparowhawk** [comment]
fixed the other error
1:36
it had to do with us calling getQueryData for folder messages on the view message screen for read receipts instead of using the RQ hook. It wasn’t a problem until this gave us the possibility of having not already called the folder messages api before entering this screen
1:36
so I updated the usage to use the RQ hook
1:36
and it no longer crashes

**Comment by rbontrager** [comment]
I've created a ticket for partial errors given in appointments, claims, and appeals [here](https://app.zenhub.com/workspaces/va-mobile-60f1a34998bc75000f2a489f/issues/gh/department-of-veterans-affairs/va-mobile-app/10112). I went through all the api calls on the list and verified that the network options match what I was seeing in staging using charles. I went through all the api calls and verified that changing the backend override changes the title, body, telephone, and refreshable state of the error. Finally I've gone through and run a spot check of the various response error codes using the other codes option for the various api calls. Approved by QA.
